kubernetes-server-mock/src/main/java/io/fabric8/kubernetes/client/server/mock/KubernetesMockServer.java

```diff
@@ -109,7 +110,26 @@ public NamespacedKubernetesClient createClient() {
     return defaultKubernetesClient;
   }
 
-  void setUnsupported(String[] unsupported) {
+  /**
+   * Used to exclude support for the given apiGroups.
+   * <br>
+   * Each is a simple expression of the form: group[/version]
+   * <br>
+   * where * is a wildcard.
+   * <br>
+   * For example to exclude all openshift support, you would specify
+   * openshift.io
+   * <br>
+   * To exclude a specific apiVersion, you would fully specify
+   * route.openshift.io/v1
+   * <p>
+   * NOTE this affects calls to {@link Client#hasApiGroup(String, boolean)}
+   * and {@link Client#supports(Class)}. Other calls to get the full root path or other
+   * api group metadata will not return valid results in mock scenarios.
+   *
+   * @param unsupported apiGroup patterns
+   */
+  public void unsupported(String... unsupported) {
```

Comment:
Are we sure we want to change the method name from setUnsupported to unsupported?

For me, this naming suggests that I might add new unsupported patterns to the ones I might have already added previously.

Reply:
Sorry, I had copied straight from the annotation.